warehouse/accounts/views.py

```diff
@@ -109,6 +110,16 @@ def login(request, redirect_field_name=REDIRECT_FIELD_NAME, _form_class=LoginFor
         redirect_field_name, request.GET.get(redirect_field_name)
     )
 
+    mfa = request.POST.get("mfa", request.GET.get("mfa"))
+    try:
+        with_mfa = bool(int(mfa))
+    except (ValueError, TypeError):
+        with_mfa = False
+
+    # Upgrade to MFA form
+    if with_mfa and _form_class == LoginForm:
```

Comment:
This check is a bit awkward. Basically, we have to check if `_form_class` is the default value set in the keyword argument. If we don't then testing becomes difficult because passing in a stubbed `form_class` will be overwritten when we set it to `LoginWithMfaForm`.

I typically don't like writing code that's been contorted to make tests pass, but I couldn't see a good way around it without a more significant refactor. I'm open to ideas though!